From fd9b0e0a5bcd0fa29507ab645719225eb7780c21 Mon Sep 17 00:00:00 2001
From: "Bernhard R. Link" <brlink@debian.org>
Date: Sun, 10 Jun 2012 18:15:17 +0200
Subject: fix 64 bit issues in lisp code

some arguments are longs where ints are expected,
so explicitly cast them.
---
 win/lisp/winlisp.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/win/lisp/winlisp.c b/win/lisp/winlisp.c
index 290a457..15f685c 100644
--- a/win/lisp/winlisp.c
+++ b/win/lisp/winlisp.c
@@ -1004,7 +1004,7 @@ generate_status_line ()
 		 lisp_int (depth(&u.uz)););
 
   lisp_list (lisp_string ("$");
-	     lisp_int (u.ugold););
+	     lisp_int ((int)u.ugold););
   lisp_list (lisp_string ("HP");
 	     lisp_int (hp););
   lisp_list (lisp_string ("HPmax");
@@ -1031,10 +1031,10 @@ generate_status_line ()
 	     lisp_int (u.ulevel););
 #ifdef EXP_ON_BOTL
   lisp_list (lisp_string ("XP");
-	     lisp_int (u.uexp););
+	     lisp_int ((int)u.uexp););
 #endif
   lisp_list (lisp_string ("T");
-	     lisp_int (moves););
+	     lisp_int ((int)moves););
 
   if (Confusion)
     lisp_list (lisp_string ("confusion"); lisp_string ("Conf"));
@@ -1932,6 +1932,6 @@ lisp_outrip(window, how)
   lisp_cmd ("outrip",
 	    lisp_int (window);	    
 	    lisp_string (plname);
-	    lisp_int (u.ugold);
+	    lisp_int ((int)u.ugold);
 	    lisp_string ("Died while trying to finish nethack-el."));
 }
